### PR TITLE
HW 7 task 1

### PR DESCRIPTION
**Pull request by limur88**

### CODE REVIEW

**Review by anatoly-karpovich — DISMISSED** [review]
Approved.
Highly recommend to use callback functions inside array methods)

**Review by VladimirLQA — DISMISSED** [review]

File: ArrMethSet-7/task2.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// 1. На вход функции подаётся предложение, например “I am the best AQA ever!” Преобразуйте строку 
+// таким образом, чтобы вместо каждой буквы была цифра, показывающая сколько раз эта буква встречается в 
+// предложении. Пробелы и знаки препинания оставляем без изменения. Регистр не должен играть роли.
+
+const sentence = 'I am the best AQA ever!';
+const countedLetters = [sentence.toLowerCase()].map(countLetters).flat();
```
Comment (VladimirLQA) [inline ArrMethSet-7/task2.js]:
```suggestion
const countedLetters = [sentence.toLowerCase()].map(countLetters).flat(Infinity).join('');
```

File: ArrMethSet-7/task2.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    for(let i = 0; i < phrase.length; i++){
+        if (alphabet.indexOf(phrase[i]) >= 0){
+        counted.push(phrase.split(phrase[i]).length -1);
+    }else{
+         counted.push(phrase[i]);
+        }
+    }return counted;
+}
+
+console.log(countedLetters);
```
Comment (VladimirLQA) [inline ArrMethSet-7/task2.js]:
```suggestion
function countLetters(phrase) {
  const counted = [];
  for (let i = 0; i < phrase.length; i++) {
    if (phrase[i].match(/[a-z]/g)) {
      counted.push(phrase.split(phrase[i]).length - 1);
    } else {
      counted.push(phrase[i]);
    }
  }
  return counted;
}

console.log(countedLetters);
```

File: ArrMethSet-7/task2.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+// 4. У вас есть массив со скобками, предположим [ ‘(‘, ‘)’, ‘(‘, ‘)’, ‘)’], количество 
+// элементов и последовательность может быть разной.
+//   Нужно выяснить, у каждой ли скобки есть соответствующая пара (открывающая и закрывающая).
+//   Усложнение: в массиве могут быть вложены еще массивы на разной глубине. 
+//   (ПОПРОБУЙТЕ МЕТОД .flat(), изучите как он работает с разными глубинами вложенности)
+//   Вернуть нужно всё также есть ли у каждой скобочки соответствующая пара. 
+//   Пример:
+
+const arr = [[['(']], ')', ')', '(', ')', ')', ['(', ['('], [')']]];
```
Comment (VladimirLQA) [inline ArrMethSet-7/task2.js:61]: Where is the function ? )

**Review by VladimirLQA — DISMISSED** [review]

**Review by anatoly-karpovich — APPROVED** [review]

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**